code/modules/mob/living/simple_animal/hostile/zombie.dm

```diff
@@ -83,8 +83,12 @@
 		qdel(src)
 		return
 	src << "<span class='userdanger'>You're down, but not quite out. You'll be back on your feet within a minute or two.</span>"
-	spawn(rand(600,900))
+	spawn(rand(300,400))
 		if(src)
+			for(var/mob/dead/observer/ghost in player_list)
+				if(src.real_name == ghost.real_name)
```

Comment:
This doesn't mean you get to enter someone else's corpse though. Ghosts have a link to their old mind, which is what allows reenter_corpse() to work. Having the same name as someone just means you may be forced to reenter YOUR old body, but not theirs.